docs/machine-learning/tutorials/text-classification-tf.md

```diff
@@ -0,0 +1,253 @@
+
+   ![sentiment_model directory contents](./media/text-classification-tf/sentiment-model-files.png)
+
+3. In Solution Explorer, right-click each of the files in the `sentiment_model` directory and subdirectory and select **Properties**. Under **Advanced**, change the value of **Copy to Output Directory** to **Copy if newer**.
+
+### Add using statements and global variables
+
+1. Add the following additional `using` statements to the top of the *Program.cs* file:
+
+   [!code-csharp[AddUsings](../../../samples/machine-learning/tutorials/TextClassificationTF/Program.cs#AddUsings "Add necessary usings")]
+
+1. Create two global variables right above the `Main` method to hold the recently downloaded dataset file path and the saved model file path:
+
+   [!code-csharp[DeclareGlobalVariables](../../../samples/machine-learning/tutorials/TextClassificationTF/Program.cs#DeclareGlobalVariables "Declare global variables")]
+
+    * `_modelPath` is the file path of the trained model.
+    * `FeatureLength` is the length of the integer feature array that the model is expecting.
+
+### Model the data
+
+Movie reviews are free form text. Your application converts the text into the input format expected by the model in a number of discrete stages.
+
+The first is to split the text into separate words and use the provided mapping file to map each word onto an integer encoding. The result of this transformation is a variable length integer array with a length corresponding to the number of words in the sentence.
+
+|Property| Value|Type|
+|-------------|-----------------------|------|
+|ReviewText|this film is really good|string|
+|VariableLengthFeatures|14,22,9,66,78,... |int[]|
+
+The variable length feature array is then resized to a fixed length of 600. This is the length that the TensorFlow model expects.
+
+|Property| Value|Type|
+|-------------|-----------------------|------|
+|ReviewText|this film is really good|string|
+|VariableLengthFeatures|14,22,9,66,78,... |int[]|
+|Features|14,22,9,66,78,... |int[600]|
+
+1. Create a class for your input data, after the `Main` method:
+
+    [!code-csharp[MovieReviewClass](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#MovieReviewClass "Declare movie review type")]
+
+    The input data class, `MovieReview`, has a `string` for user comments (`ReviewText`).
+
+1. Create a class for the variable length features, after the `Main` method:
+
+    [!code-csharp[VariableLengthFeatures](~/samples/machine-learning/tutorials/TextClassificationTF/Program.cs#VariableLengthFeatures "Declare variable length features type")]
+
+    The `VariableLengthFeatures` property has a [VectorType](xref:Microsoft.ML.Data.VectorTypeAttribute.%23ctor%2A) attribute to designate the vector type.  All of the vector elements must be the same type. In data sets with a large number of columns, loading multiple columns as a single vector reduces the number of data passes when you apply data transformations.
```

Comment:
```suggestion
    The `VariableLengthFeatures` property has a [VectorType](xref:Microsoft.ML.Data.VectorTypeAttribute.%23ctor%2A) attribute to designate it as a vector.  All of the vector elements must be the same type. In data sets with a large number of columns, loading multiple columns as a single vector reduces the number of data passes when you apply data transformations.
```

Reply:
Trying to think of a better way to rephrase that last sentence. This is form the Load Data How-To. 

> operating on the individual columns causes an impact on training speed.

In this case there's no training involved though...

Reply:
I've changed it to your wording. I even had my doubts about including this sentence. Not sure how much it adds. Thoughts?

Reply:
I think it's good to keep it since it explains why you're doing it. I guess since the same set of transformations take place on the data the speed improvements also transfer to scoring.